images/resource-consumer/consume-cpu/consume_cpu_windows.go

```diff
@@ -36,30 +37,36 @@ func doSomething() {
 }
 
 type ProcCPUStats struct {
-	User   float64   // time spent in user mode
-	System float64   // time spent in system mode
+	User   int64     // nanoseconds spent in user mode
+	System int64     // nanoseconds spent in system mode
 	Time   time.Time // when the sample was taken
-	Total  float64   // total of all time fields
+	Total  int64     // total of all time fields (nanoseconds)
 }
 
+// Retrieves the amount of CPU time this process has used since it started.
 func statsNow(handle syscall.Handle) (s ProcCPUStats) {
 	var processInfo syscall.Rusage
 	syscall.GetProcessTimes(handle, &processInfo.CreationTime, &processInfo.ExitTime, &processInfo.KernelTime, &processInfo.UserTime)
 	s.Time = time.Now()
-	s.User = float64(processInfo.UserTime.Nanoseconds())/100000000 + float64(processInfo.UserTime.Nanoseconds())/100000000
-	s.System = float64(processInfo.KernelTime.Nanoseconds())/100000000 + float64(processInfo.KernelTime.Nanoseconds())/100000000
+	s.User = processInfo.UserTime.Nanoseconds()
+	s.System = processInfo.KernelTime.Nanoseconds()
 	s.Total = s.User + s.System
 	return s
 }
 
+// Given stats from two time points, calculates the percentage of CPU used
+// by this process in the time between the samples.
 func usageNow(first ProcCPUStats, second ProcCPUStats) float64 {
-	dT := second.Time.Sub(first.Time).Seconds()
-	return 100 * (second.Total - first.Total) / dT
+	dT := second.Time.Sub(first.Time).Nanoseconds()
+	dUsagePerCPU := (second.Total - first.Total) / float64(numCPUs)
```

Comment:
Compilation error:

    .\consume_cpu_windows.go:61:47: invalid operation: (second.Total - first.Total) / float64(numCPUs) (mismatched types int64 and float64)

You could put here:

    dUsagePerCPU := float64(second.Total - first.Total) / float64(numCPUs)